tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -453,86 +426,20 @@ private static void StartDiagnosticManager()
         }
 #endif
 
-        private static void InitLiveDebugger(Tracer tracer)
+        private static void InitializeDebugger()
         {
-            var settings = tracer.Settings;
-            var debuggerSettings = DebuggerSettings.FromDefaultSource();
-
-            if (!settings.IsRemoteConfigurationAvailable)
-            {
-                // live debugger requires RCM, so there's no point trying to initialize it if RCM is not available
-                if (debuggerSettings.Enabled)
-                {
-                    Log.Warning("Live Debugger is enabled but remote configuration is not available in this environment, so live debugger cannot be enabled.");
-                }
-
-                tracer.TracerManager.Telemetry.ProductChanged(TelemetryProductType.DynamicInstrumentation, enabled: false, error: null);
-                return;
-            }
-
-            // Service Name must be lowercase, otherwise the agent will not be able to find the service
-            var serviceName = DynamicInstrumentationHelper.ServiceName;
-            var discoveryService = tracer.TracerManager.DiscoveryService;
-
-            Task.Run(
+            _ = Task.Run(
                 async () =>
                 {
-                    // TODO: LiveDebugger should be initialized in TracerManagerFactory so it can respond
-                    // to changes in ExporterSettings etc.
-
                     try
                     {
-                        var sw = Stopwatch.StartNew();
-                        var isDiscoverySuccessful = await WaitForDiscoveryService(discoveryService).ConfigureAwait(false);
-                        TelemetryFactory.Metrics.RecordDistributionSharedInitTime(MetricTags.InitializationComponent.DiscoveryService, sw.ElapsedMilliseconds);
-
-                        if (isDiscoverySuccessful)
-                        {
-                            var liveDebugger = LiveDebuggerFactory.Create(discoveryService, RcmSubscriptionManager.Instance, settings, serviceName, tracer.TracerManager.Telemetry, debuggerSettings, tracer.TracerManager.GitMetadataTagsProvider);
-
-                            Log.Debug("Initializing live debugger.");
-
-                            await InitializeLiveDebugger(liveDebugger).ConfigureAwait(false);
-                        }
+                        await DebuggerManager.Instance.UpdateDynamicConfiguration().ConfigureAwait(false);
                     }
                     catch (Exception ex)
                     {
-                        Log.Error(ex, "Error initializing live debugger.");
+                        Log.Error(ex, "Error initializing debugger");
                     }
-                });
-        }
-
-        // /!\ This method is called by reflection in the SampleHelpers
-        // If you remove it then you need to provide an alternative way to wait for the discovery service
```

Comment:
> If you remove it then you need to provide an alternative way to wait for the discovery service

What this handled in this PR?

Reply:
Never mind, I see it moved to `tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs`